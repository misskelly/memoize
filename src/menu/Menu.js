import React, { Component } from 'react';
import Button from '../button/Button';
import { filterDeck } from '../lib/utils';



export default class Menu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      student: 'Friend',
    }
    this.selectDeck = this.selectDeck.bind(this);
  }

  updateStudent = (e) => {
    let name = e.target.value;
    this.setState({
      student: name
    })
  } 
  
  selectDeck = (chosenDeck) => {
    const { allCards, updateDeck } = this.props;
    const updatedDeck = filterDeck(chosenDeck, allCards);
    updateDeck(updatedDeck);
    console.log('chosen:', chosenDeck, 'updated:',updatedDeck)
  }
  
  handleSubmit = (e) => {
    const { student } = this.state;
    e.preventDefault();
    console.log(this.state)
    const greeting = document.querySelector('.greetingHeading');
    const input = document.querySelector('.nameInput');
    greeting.innerText = `Hello ${student}!`;
    input.value = '';
  }
  
  //  TODO : update student method
  
  render () {
    const { allCards, updateDeck, getRandomCard } = this.props;
    return (
      <section className='mainMenu'>
        <h3 className='greetingHeading'>Hello Friend!</h3>
        <form className='menuForm' onSubmit={this.handleSubmit}>
          <label  className='nameInputLabel'
                  htmlFor='nameInput'>
                  Student Name
          </label>
          <input  type='text'
                  id='nameInput'
                  className='nameInput' placeholder='Your name here.....'
                  onChange={this.updateStudent}/>
          <button type='submit' 
                  className='updateNameButton'
                  autoComplete = "fname" 
                  autoFocus>
                  
          </button>
          <h4 className='chooseDeckHeading'>
            Which terms would you like to review?
          </h4>

          <Button type='basicsDeckBtn'
                  buttonText='React Basics'
                  label=''
                  selectDeck={this.selectDeck}
                  getRandomCard={getRandomCard}
                  updateDeck={updateDeck}/>
                  
          <Button type='testingDeckBtn'
                  buttonText='React Testing'
                  label=''
                  selectDeck={this.selectDeck}/>
          <Button type='previousDeck'
                  buttonText='Keep Going'
                  label=''
                  getRandomCard={getRandomCard}
                  updateDeck={updateDeck}
                  // TODO: local Storage!!!!                  
                  // selectDeck={storedDeck}
                  />
          <Button type='fullDeck'
                  buttonText='Everything'
                  label=''
                  getRandomCard={getRandomCard}
                  allTerms={this.updateDeck}
                  allCards={allCards}/>

        </form>
      
      </section>
    )
  }


}