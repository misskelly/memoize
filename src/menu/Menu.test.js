import React from 'react';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('Menu', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Menu />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});