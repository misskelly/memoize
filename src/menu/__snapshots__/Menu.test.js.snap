// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="menuForm"
      >
        <label
          className="nameInputLabel"
          htmlFor="nameInput"
        >
          Student
        </label>
        <input
          className="nameInput"
          id="nameInput"
          placeholder="Your name here....."
          type="text"
        />
        <h3
          className="chooseDeckHeading"
        >
          Which terms would you like to review?
        </h3>
      </form>,
      "className": "mainMenu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="nameInputLabel"
            htmlFor="nameInput"
          >
            Student
          </label>,
          <input
            className="nameInput"
            id="nameInput"
            placeholder="Your name here....."
            type="text"
          />,
          <h3
            className="chooseDeckHeading"
          >
            Which terms would you like to review?
          </h3>,
        ],
        "className": "menuForm",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Student",
            "className": "nameInputLabel",
            "htmlFor": "nameInput",
          },
          "ref": null,
          "rendered": "Student",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "nameInput",
            "id": "nameInput",
            "placeholder": "Your name here.....",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Which terms would you like to review?",
            "className": "chooseDeckHeading",
          },
          "ref": null,
          "rendered": "Which terms would you like to review?",
          "type": "h3",
        },
      ],
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="menuForm"
        >
          <label
            className="nameInputLabel"
            htmlFor="nameInput"
          >
            Student
          </label>
          <input
            className="nameInput"
            id="nameInput"
            placeholder="Your name here....."
            type="text"
          />
          <h3
            className="chooseDeckHeading"
          >
            Which terms would you like to review?
          </h3>
        </form>,
        "className": "mainMenu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="nameInputLabel"
              htmlFor="nameInput"
            >
              Student
            </label>,
            <input
              className="nameInput"
              id="nameInput"
              placeholder="Your name here....."
              type="text"
            />,
            <h3
              className="chooseDeckHeading"
            >
              Which terms would you like to review?
            </h3>,
          ],
          "className": "menuForm",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Student",
              "className": "nameInputLabel",
              "htmlFor": "nameInput",
            },
            "ref": null,
            "rendered": "Student",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "nameInput",
              "id": "nameInput",
              "placeholder": "Your name here.....",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Which terms would you like to review?",
              "className": "chooseDeckHeading",
            },
            "ref": null,
            "rendered": "Which terms would you like to review?",
            "type": "h3",
          },
        ],
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
